src/runtime/description-formatter.ts

```diff
@@ -11,42 +11,35 @@
 import {assert} from '../platform/assert-web.js';
 import {ParticleSpec} from './particle-spec.js';
 import {Arc} from './arc.js';
-import {Recipe} from './recipe/recipe.js';
 import {Handle} from './recipe/handle.js';
 import {Particle} from './recipe/particle.js';
 import {HandleConnection} from './recipe/handle-connection.js';
-import {StorageProviderBase} from './storage/storage-provider-base.js';
-import {EntityType, CollectionType, BigCollectionType, InterfaceType} from './type.js';
-import {Relevance} from './relevance';
+import {CollectionType, BigCollectionType, InterfaceType} from './type.js';
 
-export type ParticleDescription = {_particle: Particle, pattern?: string, _connections: {[index: string]: HandleDescription}, _rank?: number};
-export type HandleDescription = {pattern: string, _handleConn: HandleConnection, _store: StorageProviderBase};
+export type ParticleDescription = {
+  _particle: Particle,
+  pattern?: string,
+  _connections: {[index: string]: HandleDescription},
+  _rank?: number
+};
+export type HandleDescription = {pattern: string, _handleConn: HandleConnection, value: DescriptionValue};
+export type DescriptionValue = {entityValue?: string|{}, valueDescription?: string, collectionValues?: string[], bigCollectionValues?: string[], interfaceValue?: string | {}};
 
 export type CombinedDescriptionsOptions = {skipFormatting?: boolean};
 
 export class DescriptionFormatter {
-  private arc: Arc;
-  private relevance: Relevance | null = null;
-  private particleDescriptions = <ParticleDescription[]>[];
   private seenHandles: Set<Handle> = new Set();
-  
   seenParticles: Set<Particle> = new Set();
-  
   excludeValues = false;
-  
-  constructor(arc: Arc, relevance?: Relevance) {
-    assert(arc, `Arc is mandatory`);
-    this.arc = arc;
-    this.relevance = relevance;
-  }
 
-  async getDescription(recipe: {patterns: string[], particles: Particle[]}) {
-    await this._updateDescriptionHandles();
+  constructor(private readonly particleDescriptions = <ParticleDescription[]>[],
```

Comment:
Shane had some opinions on whether we should instantiate properties in the constructor like this.

I'm fine with it fwiw, as it's a more compact representation...

Reply:
I relied on some existing examples of ctors, 
but I can update it in a follow up PR, if it is an undesirable way.